src/tests/ftest/rebuild/container_rf_enforce.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from daos_utils import DaosCommand
+from general_utils import DaosTestError
+from container_redundancy import RbldContRedundancyFactor
+import re
+
+class RbldContRedundancyFactorEnforce(RbldContRedundancyFactor):
+    # pylint: disable=too-many-ancestors
+    """Test Container redundancy factor enforcement with rebuild.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a Rebuild Container RF with ObjClass Write object."""
+        super().__init__(*args, **kwargs)
+        self.mode = None
+        self.daos_cmd = None
+
+    def create_test_container_and_write_obj(self, positive_test=True):
+        """Create a container and write objects
+           for positive testcase, enable the exception with write objects
+           for negative testcase, disable the exception with write objects
+           and expecting failure with RC: -1003.
+
+        Args:
+            positive_test (Bool, optional): expected container test
+            write_object pass for positive test.
+        """
+        der_inval = "RC: -1003"
+        self.log.info(
+            "==>(1)Create pool and container with redundant factor,"
+            " start background IO object write")
+        self.container.create()
+        if positive_test:
+            self.container.write_objects(
+                self.inputs.rank.value[0], self.inputs.object_class.value)
+        else:
+            try:
+                self.container.write_objects_no_except(
+                    self.inputs.rank.value[0], self.inputs.object_class.value)
+                self.fail("#RF and traffic with invallid object_class Passed,"
+                          " expecting failed.")
+            except DaosTestError as error:
+                self.log.info(error)
+                if der_inval in str(error):
+                    self.log.info(" ==excepting DaosApiError -1003 found.")
+                else:
+                    self.fail("#Test failed RC: -1003 not found")
+
+    def execute_rebuild_test(self, create_container=True):
```

Comment:
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.